packages/fiori/src/UploadCollectionItem.ts

```diff
@@ -0,0 +1,512 @@
+	 *
+	 * @type {Node[]}
+	 * @slot
+	 * @name sap.ui.webc.fiori.UploadCollectionItem.prototype.default
+	 * @public
+	 */
+
+	static i18nFioriBundle: I18nBundle;
+
+	static get styles() {
+		return [ListItem.styles, UploadCollectionItemCss];
+	}
+
+	static get template() {
+		return UploadCollectionItemTemplate;
+	}
+
+	static get dependencies() {
+		return [
+			...ListItem.dependencies,
+			Button,
+			Input,
+			Link,
+			Label,
+			ProgressIndicator,
+		];
+	}
+
+	static async onDefine() {
+		[UploadCollectionItem.i18nFioriBundle] = await Promise.all([
+			getI18nBundle("@ui5/webcomponents-fiori"),
+			super.onDefine(),
+		]);
+	}
+
+	onBeforeRendering() {
+		// In the base class the item can become "actionable",
+		// that's why we are overriding this method.
+	}
+
+	/**
+	 * @override
+	 */
+	async onDetailClick() {
+		super.onDetailClick();
+		this._editing = true;
+
+		await this._initInputField();
+	}
+
+	async _initInputField() {
+		await renderFinished();
+
+		const inp = this.shadowRoot!.querySelector<Input>("#ui5-uci-edit-input")!;
+		inp.value = this._fileNameWithoutExtension;
+
+		await renderFinished();
+
+		const inpFocusDomRef = inp.getFocusDomRef() as HTMLInputElement;
+		if (inpFocusDomRef) {
+			inpFocusDomRef.focus();
+			inpFocusDomRef.setSelectionRange(0, this._fileNameWithoutExtension.length);
+		}
+	}
+
+	_onDetailKeyup(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			this.onDetailClick();
+		}
+	}
+
+	_onInputFocusin(e: FocusEvent) {
+		// prevent focusing the whole upload collection item.
+		e.stopPropagation();
+	}
+
+	_onInputKeyDown(e: KeyboardEvent) {
+		if (isEscape(e)) {
+			this._onRenameCancel(e);
+		} else if (isEnter(e)) {
+			this._onRename();
+		} else if (isSpace(e)) {
+			e.stopImmediatePropagation();
+		}
+	}
+
+	_onRename() {
+		const inp = this.shadowRoot!.querySelector<Input>("#ui5-uci-edit-input")!;
+		this.fileName = inp.value + this._fileExtension;
+		this.fireEvent("rename");
+
+		this._editing = false;
+		this._focus();
+	}
+
+	_onRenameKeyup(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			this._onRename();
+		}
+	}
+
+	async _onRenameCancel(e: KeyboardEvent) {
+		this._editing = false;
+
+		if (isEscape(e)) {
+			await renderFinished();
+			this.shadowRoot!.querySelector<Button>(`#${this._id}-editing-button`)!.focus();
+		} else {
+			this._focus();
+		}
+	}
+
+	_onRenameCancelKeyup(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			this._onRenameCancel(e);
+		}
+	}
+
+	_focus() {
+		this.fireEvent("_focus-requested");
+	}
+
+	_onFileNameClick() {
+		this.fireEvent("file-name-click");
+	}
+
+	_onRetry() {
+		this.fireEvent("retry");
+	}
+
+	_onRetryKeyup(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			this._onRetry();
+		}
+	}
+
+	_onTerminate() {
+		this.fireEvent("terminate");
+	}
+
+	_onTerminateKeyup(e: KeyboardEvent) {
+		if (isSpace(e)) {
+			this._onTerminate();
+		}
+	}
+
+	getFocusDomRef() {
+		return this.getDomRef();
+	}
+
+	get list() {
```

Comment:
do we really need that getter? I wasn't able to find its usage.

Reply:
Looks like old code. Removed it.